test/rdoc/test_rdoc_store.rb

```diff
@@ -281,6 +281,26 @@ def test_find_c_enclosure_from_cache_legacy
     assert_nil @s.find_c_enclosure('cObject')
   end
 
+  def test_resolve_c_references
+    # first parse a child that references an unknown parent
+    c_file1 = @s.add_file 'ext1.c'
+    c_file1.add_class RDoc::NormalClass, 'Child', 'cExternParent'
+
+    # then parse the parent and register the C variable name as a C enclosure
+    c_file2 = @s.add_file 'ext2.c'
+    parent = c_file2.add_class RDoc::NormalClass, 'Parent', 'rb_cObject'
+
+    @s.add_c_enclosure('cExternParent', parent)
+
+    # at this point, the child's superclass is still the name of the C variable
+    assert_equal("cExternParent", @s.classes_hash['Child'].superclass)
+
+    @s.resolve_c_references
+
+    # now the ancestor tree correctly references the NormalClass objects
+    assert_equal(parent, @s.classes_hash['Child'].superclass)
```

Comment:
(`#superclass` can return either string or `RDoc::NormalClass` really hurts my brain 🙈)

Reply:
😬

Reply:
And of course the bug in #1221 is related to this confusion. 🤦